pkg/server/commands/check_command.go

```diff
@@ -27,25 +27,83 @@ const (
 	defaultMaxConcurrentReadsForCheck = math.MaxUint32
 )
 
+// CheckCommand interface wraps the Execute method for performing a check operation.
+type CheckCommand interface {
+	Execute(ctx context.Context) (*graph.ResolveCheckResponse, *graph.ResolveCheckRequestMetadata, error)
+}
+
 type CheckQuery struct {
+	params                     CheckCommandParams
 	logger                     logger.Logger
 	checkResolver              graph.CheckResolver
-	typesys                    *typesystem.TypeSystem
-	datastore                  storage.RelationshipTupleReader
 	sharedCheckResources       *shared.SharedDatastoreResources
 	cacheSettings              config.CacheSettings
 	maxConcurrentReads         uint32
 	shouldCacheIterators       bool
 	datastoreThrottleThreshold int
 	datastoreThrottleDuration  time.Duration
+	datastoreWithTupleCache    *storagewrappers.RequestStorageWrapper
 }
 
+var _ CheckCommand = (*CheckQuery)(nil)
```

Comment:
Fun that _all_ methods within the command pkg are actually named `XQuery` and only Check is called Command... :)